TitanRobotics2024/src/main/java/frc/robot/Subsystem/Control.java

```diff
@@ -23,21 +36,166 @@ public Control()
+        ramp = Ramp.getInstance();
+        inversion = false;
     }
 
     public void teleopControl()
     {
         forward = -driverController.getStick(ButtonMap.XboxLEFTSTICKY) * (Math.abs(driverController.getStick(ButtonMap.XboxLEFTSTICKY)));
         turn = -driverController.getStick(ButtonMap.XboxRIGHTSTICKX);
+
+        if (driverController.debounceB()){
+            inversion = !inversion;
+            SmartDashboard.putBoolean("inversion", inversion);
+        }
+
+        if (inversion){
+
+            forward = -forward;
+        }
+
+        //limelightControl();
+
         driveBase.drive(forward, turn);
+
+        climberControl();
+        manipulatorControl();
+    }
+
+    private void limelightControl()
+    {
+        targeting.setAlliance("blue"); // Change depending on alliance
+                                       // for upcoming match.
+                                       // Failure to change this will
+                                       // cause you to target the
+                                       // wrong AprilTags when using
+                                       // lock on buttons.
+        if (driverController.debounceSTART())
+        {
+            System.out.println("pressed");
+            if (limelight.getPipeline() == 0)
+            {
+                limelight.setPipeline(1);
+            }
+            else
+            {
+                limelight.setPipeline(0);
+            }
+        }
+
+        if (limelight.getPipeline() == 0)
+        {
+            if (driverController.getButton(ButtonMap.XboxX))
+            {
+                targeting.setTarget("Amp");
+                turn = targeting.aprilTaglockOn();
+                System.out.println("Locking on to Amp");
+            }
+
+            if (driverController.getButton(ButtonMap.XboxA))
+            {
+                targeting.setTarget("Source");
+                turn = targeting.aprilTaglockOn();
+                System.out.println("Locking on to Source");
+            }
+
+            if (driverController.getButton(ButtonMap.XboxB))
+            {
+                targeting.setTarget("Stage");
+                turn = targeting.aprilTaglockOn();
+                System.out.println("Locking on to Stage");
+            }
+
+            else
+            {
+                targeting.setTarget("none");
+            }
+        }
+
+        if (limelight.pipeline == 1)
+        {
+            if (driverController.getButton(ButtonMap.XboxY))
+            {
+                turn = targeting.otherLockOn();
+                System.out.println("Locking On to Note");
+            }
+        }
+
+        /*if (driverController.getButton(ButtonMap.XboxRIGHTBumper))
+        {
+            forward = targeting.follow();
+            turn = targeting.otherLockOn();
+        }*/
+    }
+
+    private void climberControl()
+    {
+
+        // if (operatorController.getButton(ButtonMap.XboxY))
```

Comment:
Remove commented out code

Reply:
will use when have encoders

Reply:
You don't support encoders, don't include in main